### ISSUE TITLE
Failed when requiring memcached

### ISSUE THREAD

**Issue by ekho** [tarantool/memcached#11]
[Vagrantfile](https://gist.github.com/ekho/56b6329c0be404ccaf37#file-vagrantfile_luarocks)

Installed with luarocks.

When trying step by step `Usage` instructions from readme (with small change) and got error:

```
vagrant@vagrant-ubuntu-trusty-64:~$ tarantool
tarantool: version 1.6.6-269-gfe1519e
type 'help' for interactive help
tarantool> box.cfg{slab_alloc_arena = 0.1}
2015-11-02 20:54:24.379 [27246] main/101/interactive C> version 1.6.6-269-gfe1519e
2015-11-02 20:54:24.379 [27246] main/101/interactive C> log level 5
2015-11-02 20:54:24.379 [27246] main/101/interactive I> mapping 107374184 bytes for tuple arena...
2015-11-02 20:54:24.382 [27246] main/101/interactive I> initializing an empty data directory
2015-11-02 20:54:24.389 [27246] snapshot/101/main I> creating `./00000000000000000000.snap.inprogress'
2015-11-02 20:54:24.390 [27246] snapshot/101/main I> saving snapshot `./00000000000000000000.snap.inprogress'
2015-11-02 20:54:24.391 [27246] snapshot/101/main I> done
2015-11-02 20:54:24.395 [27246] main/101/interactive I> ready to accept requests
---
...

tarantool> local memcached = require('memcached')
---
- error: '/home/vagrant/.luarocks/share/lua/5.1/memcached/init.lua:11: bad argument
    #1 to ''load'' (string expected, got nil)'
...

tarantool> 
```

11: `ffi.load(package.searchpath('memcached.internal', package.cpath), true)`

**Comment by bigbes** [comment]
Can you show me log of luarocks?

**Comment by ekho** [comment]
```
==> default: Using https://raw.githubusercontent.com/tarantool/memcached/master/memcached-scm-1.rockspec... switching to 'build' mode
==> default: Cloning into 'memcached'...
==> default: gcc -O2 -fPIC -I/usr/include/lua5.1 -c memcached/internal/memcached.c -o memcached/internal/memcached.o -I/usr/include/tarantool -I/usr/include/small -Ithird_party
==> default: memcached/internal/memcached.c: In function ‘memcached_loop_error’:
==> default: memcached/internal/memcached.c:223:23: warning: initialization discards ‘const’ qualifier from pointer target type [enabled by default]
==> default:   box_error_t *error = box_error_last();
==> default:                        ^
==> default: memcached/internal/memcached.c: In function ‘memcached_expire_start’:
==> default: memcached/internal/memcached.c:411:15: warning: assignment makes pointer from integer without a cast [enabled by default]
==> default:   expire_fiber = fiber_new(name, memcached_expire_loop);
==> default:                ^
==> default: gcc -O2 -fPIC -I/usr/include/lua5.1 -c memcached/internal/memcached_constants.c -o memcached/internal/memcached_constants.o -I/usr/include/tarantool -I/usr/include/small -Ithird_party
==> default: gcc -O2 -fPIC -I/usr/include/lua5.1 -c memcached/internal/memcached_layer.c -o memcached/internal/memcached_layer.o -I/usr/include/tarantool -I/usr/include/small -Ithird_party
==> default: gcc -O2 -fPIC -I/usr/include/lua5.1 -c memcached/internal/memcached_network.c -o memcached/internal/memcached_network.o -I/usr/include/tarantool -I/usr/include/small -Ithird_party
==> default: memcached/internal/memcached_network.c: In function ‘iobuf_mempool_create’:
==> default: memcached/internal/memcached_network.c:33:62: warning: passing argument 2 of ‘mempool_create’ makes pointer from integer without a cast [enabled by default]
==> default:   mempool_create(&ibuf_pool, cord_slab_cache(), sizeof(struct ibuf));
==> default:                                                               ^
==> default: In file included from memcached/internal/memcached_network.c:12:0:
==> default: /usr/include/small/mempool.h:213:1: note: expected ‘struct slab_cache *’ but argument is of type ‘int’
==> default:  mempool_create(struct mempool *pool, struct slab_cache *cache,
==> default:  ^
==> default: memcached/internal/memcached_network.c:34:62: warning: passing argument 2 of ‘mempool_create’ makes pointer from integer without a cast [enabled by default]
==> default:   mempool_create(&obuf_pool, cord_slab_cache(), sizeof(struct obuf));
==> default:                                                               ^
==> default: In file included from memcached/internal/memcached_network.c:12:0:
==> default: /usr/include/small/mempool.h:213:1: note: expected ‘struct slab_cache *’ but argument is of type ‘int’
==> default:  mempool_create(struct mempool *pool, struct slab_cache *cache,
==> default:  ^
==> default: memcached/internal/memcached_network.c: In function ‘ibuf_new’:
==> default: memcached/internal/memcached_network.c:50:22: warning: passing argument 2 of ‘ibuf_create’ makes pointer from integer without a cast [enabled by default]
==> default:   ibuf_create((struct ibuf *)ibuf, cord_slab_cache(), iobuf_readahead);
==> default:                       ^
==> default: In file included from memcached/internal/memcached_network.c:13:0:
==> default: /usr/include/small/ibuf.h:72:1: note: expected ‘struct slab_cache *’ but argument is of type ‘int’
==> default:  ibuf_create(struct ibuf *ibuf, struct slab_cache *slabc, size_t start_capacity);
==> default:  ^
==> default: memcached/internal/memcached_network.c: In function ‘obuf_new’:
==> default: memcached/internal/memcached_network.c:59:22: warning: passing argument 2 of ‘obuf_create’ makes pointer from integer without a cast [enabled by default]
==> default:   obuf_create((struct obuf *)obuf, cord_slab_cache(), iobuf_readahead);
==> default:                       ^
==> default: In file included from memcached/internal/memcached_network.c:14:0:
==> default: /usr/include/small/obuf.h:108:1: note: expected ‘struct slab_cache *’ but argument is of type ‘int’
==> default:  obuf_create(struct obuf *buf, struct slab_cache *slabc, size_t start_capacity);
==> default:  ^
==> default: memcached/internal/memcached_network.c: In function ‘mnet_writev’:
==> default: memcached/internal/memcached_network.c:98:17: error: ‘COIO_WRITE’ undeclared (first use in this function)
==> default:    coio_wait(fd, COIO_WRITE, TIMEOUT_INFINITY);
==> default:                  ^
==> default: memcached/internal/memcached_network.c:98:17: note: each undeclared identifier is reported only once for each function it appears in
==> default: memcached/internal/memcached_network.c: In function ‘mnet_write’:
==> default: memcached/internal/memcached_network.c:118:17: error: ‘COIO_WRITE’ undeclared (first use in this function)
==> default:    coio_wait(fd, COIO_WRITE, TIMEOUT_INFINITY);
==> default:                  ^
==> default: memcached/internal/memcached_network.c: In function ‘mnet_read_ahead’:
==> default: memcached/internal/memcached_network.c:142:17: error: ‘COIO_READ’ undeclared (first use in this function)
==> default:    coio_wait(fd, COIO_READ, TIMEOUT_INFINITY);
==> default:                  ^
==> default: No existing manifest. Attempting to rebuild...
==> default: Updating manifest for /home/vagrant/.luarocks/lib/luarocks/rocks
==> default: 
==> default: memcached scm-1 is now built and installed in /home/vagrant/.luarocks/ (license: BSD)
```

**Comment by bigbes** [comment]
@ekho, so it seems that you've install tarantool from `master` repository instead of `1.6`. Try to replace `master` with `1.6` in your `source.list` file.

**Comment by bigbes** [comment]
It seems like problem now - we have two repositories, `master` and `1.6` - you must install `libsmall` from `master` repository, but `tarantool` and `tarantool-dev` from `1.6` repository. But it's only for a couple of weeks, until 1.6 will be merged into master branch in git.

**Comment by ekho** [comment]
Where is no `libsmall` and `libsmall-dev` in `1.6` repo =))))

**Comment by ekho** [comment]
Thanks. I'll try next time.

**Comment by ekho** [comment]
@bigbes
I installed libsmall and libsmall-dev from master and tarantool from 1.6.
And got new error =)))))

```
tarantool> box.cfg{slab_alloc_arena = 0.1}
2015-11-03 10:51:53.305 [6008] main/101/interactive C> version 1.6.7-363-g838bb33
2015-11-03 10:51:53.305 [6008] main/101/interactive C> log level 5
2015-11-03 10:51:53.305 [6008] main/101/interactive I> mapping 107374184 bytes for tuple arena...
2015-11-03 10:51:53.306 [6008] main/101/interactive I> initializing an empty data directory
2015-11-03 10:51:53.311 [6008] snapshot/101/main I> creating `./00000000000000000000.snap.inprogress'
2015-11-03 10:51:53.312 [6008] snapshot/101/main I> saving snapshot `./00000000000000000000.snap.inprogress'
2015-11-03 10:51:53.312 [6008] snapshot/101/main I> done
2015-11-03 10:51:53.315 [6008] main/101/interactive I> ready to accept requests
---
...

tarantool> local memcached = require('memcached')
---
...

tarantool> local instance = memcached.create('testinstance', '0.0.0.0:11211')
---
- error: '[string "local instance = memcached.create(''testinstan..."]:1: attempt
    to index global ''memcached'' (a nil value)'
...

tarantool>
```

**Comment by ekho** [comment]
Where is my mistake ? DNA? =)))))

**Comment by bigbes** [comment]
You're running it from REPL interface, but `local` scope for out REPL is one line - current:)
```
tarantool> local memcached = require('memcached') -- Here you can reach `local` variable `memcached`
---
...
tarantool> local instance = memcached.create('testinstance', '0.0.0.0:11211') -- Here you can't do it :(
---
- error: '[string "local instance = memcached.create(''testinstan..."]:1: attempt
    to index global ''memcached'' (a nil value)'
...
```
Try use it in REPL without `local` scope - it might work.

**Comment by bigbes** [comment]
@ekho, is everything ok?

**Comment by ekho** [comment]
@bigbes , yes, with global vars works fine in REPL
